### ISSUE TITLE
Improve settings debug helper

### ISSUE THREAD

**Issue by fm-sys** [cgeo/cgeo#9123]
The debug settings dialog (#8432 - which I really like!) has two bugs/missing functionality:

1. You can't enter negative number values currently 
2. You can't create a preference key if it not jet exists

(3. And while we are working on it we can replace the edit icon with the ✓ icon when editing is enabled)

**Comment by moving-bits** [comment]
May I add one wish?
On closing the view, check whether there are unsaved changes, and offer to save or discard them, if there are.

**Comment by moving-bits** [comment]
For the "you can't enter negative number values currently" part look at `settings/ViewSettingsActivity#182` and add `InputType.TYPE_NUMBER_FLAG_SIGNED`.

**Comment by moving-bits** [comment]
@fm-sys 
Are you working on this (you have it assigned to you), or shall I take over?

**Comment by fm-sys** [comment]
> Are you working on this (you have it assigned to you), or shall I take over? 

I'm not really working on it. I had a short look at it some weeks ago, but I came to the conclusion that it isn't that simple to create a not existing preference key cause we don't now the correct type.
> You can't create a preference key if it not jet exists

From my side we can also leave it out, it is only a debug feature and we shouldn't "waste" too much time.

Until now I only added the number type flag. I could provide a PR for it, but that would probably take more time for both of us than if you add the line yourself while continuing with this issue.

To make a long story short: A takeover would be more than welcomed from my side!
